chain/client/src/test_utils/test_env_builder.rs

```diff
@@ -360,10 +365,15 @@ impl TestEnvBuilder {
 
     /// Internal impl to make sure runtimes are initialized.
     fn ensure_runtimes(self) -> Self {
+        let state_snapshot_enabled = self.state_snapshot_enabled;
```

Comment:
just out of curiosity, why do we need to do this allocation?

Reply:
`self` gets "consumed" in the call to `ensure_epoch_managers()` and can not be used later to access any variable. We can only access `ret` but not it's private members like `state_snapshot_enabled`